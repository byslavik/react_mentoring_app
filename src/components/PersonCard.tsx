import * as React from 'react';
import * as s from './scss/FilmCard.scss';

import { Link } from 'react-router-dom';

import { Film } from '../models/film.model';

import {SETTINGS} from './../settings';


interface PersonCardProps {
  item: Film;
  key?: number;
}

export  class PersonCard extends React.Component<PersonCardProps, any> {
  render () {
    const item:Film = this.props.item;
    const popularity:any = item.popularity;
    return (
      <article className={s.filmCard}>
        <div className={s.imageWrap}>
          <img src={SETTINGS.imgUrl + item.profile_path} alt={item.name}/>
        </div>
        <h1><Link to={`/person/${item.name}/${item.id}`}>{item.name}</Link> <span>{(Math.round(popularity * 100) / 100 )}</span></h1>
      </article>
    )
  }
}
