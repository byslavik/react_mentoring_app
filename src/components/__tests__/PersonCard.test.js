import * as React from 'react';
import {PersonCard} from '../PersonCard';
import * as renderer from 'react-test-renderer';
import {filmData} from './testData'

jest.mock('react-router-dom',()=> ({Link:'Link'}));

describe('PersonCard component', ()=>{
  it('PersonCard component shows correctly', () => {
    const tree = renderer.create(
      <PersonCard item={filmData}/>
    ).toJSON();
    expect(tree).toMatchSnapshot();
  });
})