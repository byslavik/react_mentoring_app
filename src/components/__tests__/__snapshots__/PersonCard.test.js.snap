// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PersonCard component PersonCard component shows correctly 1`] = `
<article
  className={undefined}
>
  <div
    className={undefined}
  >
    <img
      alt="Name"
      src="https://image.tmdb.org/t/p/w500//somepath.jpg"
    />
  </div>
  <h1>
    <Link
      to="/person/Name/1"
    >
      Name
    </Link>
     
    <span>
      100500
    </span>
  </h1>
</article>
`;
